stack/__main__.py

```diff
@@ -331,13 +335,98 @@ def create_group(mail: str) -> gcp.cloudidentity.Group:
-            f'sample-metadata-api-{group_prefix}-access-group-cache-secret-accessor',
-            secret_id=access_group_cache_secrets[group_prefix].id,
-            role='roles/secretmanager.secretAccessor',
-            member=f'serviceAccount:{SAMPLE_METADATA_API_SERVICE_ACCOUNT}',
-        )
+    # Sample metadata access
+    #   - 4 secrets, main-read, main-write, test-read, test-write
+    sm_groups = {}
+    for env in ('main', 'test'):
+        for rs in ('read', 'write'):
+            key = f'sample-metadata-{env}-{rs}'
+            secret_id = f'{dataset}-{key}-members-cache'
+
+            group = create_group(group_mail(dataset, key))
+            sm_groups[f'{env}-{rs}'] = group
+
+            gcp.cloudidentity.GroupMembership(
+                f'sample-metadata-group-cache-{env}-{rs}-group-membership',
+                group=group,
+                preferred_member_key=gcp.cloudidentity.GroupMembershipPreferredMemberKeyArgs(
+                    id=ACCESS_GROUP_CACHE_SERVICE_ACCOUNT
+                ),
+                roles=[gcp.cloudidentity.GroupMembershipRoleArgs(name='MEMBER')],
+                opts=pulumi.resource.ResourceOptions(depends_on=[cloudidentity]),
+            )
+
+            gcp.secretmanager.Secret(
+                f'{key}-group-cache-secret',
+                secret_id=secret_id,
+                replication=gcp.secretmanager.SecretReplicationArgs(
+                    user_managed=gcp.secretmanager.SecretReplicationUserManagedArgs(
+                        replicas=[
+                            gcp.secretmanager.SecretReplicationUserManagedReplicaArgs(
+                                location='australia-southeast1',
+                            ),
+                        ],
+                    ),
+                ),
+                opts=pulumi.resource.ResourceOptions(depends_on=[secretmanager]),
+            )
+
+            gcp.secretmanager.SecretIamMember(
+                f'{key}-api-secret-accessor',
+                secret_id=secret_id,
+                role='roles/secretmanager.secretAccessor',
+                member=f'serviceAccount:{SAMPLE_METADATA_API_SERVICE_ACCOUNT}',
+            )
+            gcp.secretmanager.SecretIamMember(
+                f'{key}-group-cache-secret-accessor',
+                secret_id=secret_id,
+                role='roles/secretmanager.secretAccessor',
+                member=f'serviceAccount:{ACCESS_GROUP_CACHE_SERVICE_ACCOUNT}',
+            )
+
+            gcp.secretmanager.SecretIamMember(
+                f'{key}-group-cache-secret-version-manager',
+                secret_id=secret_id,
+                role='roles/secretmanager.secretVersionManager',
+                member=f'serviceAccount:{ACCESS_GROUP_CACHE_SERVICE_ACCOUNT}',
+            )
+
+    # Declare access to sample-metadata API of format ({env}-{read,write})
+    sm_access_levels: List[SampleMetadataAccessorMembership] = [
+        SampleMetadataAccessorMembership(
+            name='person',
```

Comment:
(For consistency with other documentation.)

```suggestion
            name='human',
```